src/librustc_trans/trans/base.rs

```diff
@@ -2824,6 +2856,12 @@ pub fn trans_crate(tcx: &ty::ctxt, analysis: ty::CrateAnalysis) -> CrateTranslat
                             &reachable_symbols.iter().map(|x| &x[..]).collect());
     }
 
+    if sess.target.target.options.is_like_msvc &&
+       sess.crate_types.borrow().iter().any(|ct| *ct == config::CrateTypeRlib ||
+                                                 *ct == config::CrateTypeStaticlib) {
```

Comment:
We can only do this for rlibs, right? If you're producing a staticlib then on one else is going to be referencing *this crate's* globals via `dllimport` (not correctly, at least), right?